chiselFrontend/src/main/scala/Chisel/Module.scala

```diff
@@ -70,14 +78,16 @@ extends HasId {
     Port(port, if (port.dir == NO_DIR) bundleDir else port.dir)
   }
 
-  private[Chisel] def setupInParent(): this.type = _parent match {
-    case Some(p) => {
-      pushCommand(DefInvalid(io.ref)) // init instance inputs
-      clock := override_clock.getOrElse(p.clock)
-      reset := override_reset.getOrElse(p.reset)
-      this
+  private[Chisel] def setupInParent(implicit sourceInfo: SourceInfo): this.type = {
```

Comment:
This function is only ever called internally, does it need source info implicitly, or even at all?